web-registry/src/pages/MyEcocredits.tsx

```diff
@@ -94,6 +100,49 @@ const WrappedMyEcocredits: React.FC<WithBasketsProps> = ({ baskets }) => {
     }
   }, [credits, basketsWithClasses]);
 
+  const openTakeModal = (rowIndex: number): void => {
+    if (!accountAddress) return;
+    const selectedBasket = basketsWithClasses?.[rowIndex]?.basket;
+    if (selectedBasket?.basketDenom) {
+      setSelectedBasketDenom(selectedBasket.basketDenom);
+    }
+  };
+
+  const handleTakeCredits = async (values: MsgTakeValues): Promise<void> => {
```

Comment:
let's be consistent with function naming, we have `handleTakeCredits` and `basketPutSubmit`
what about `basketTakeSubmit`?

Reply:
Now `basketTakeSubmit`